### DISCUSSION TITLE
Prisma2 and dockerfile + docker-compose

### DISCUSSION THREAD

**Discussion by iurygregory-dev** [prisma/prisma#2653] [Q&A] [answered]
hi, my docker file is generating an error when executing the command:
npx prisma migrate save --experimental

error: Error: P1001: Can't reach database server at `postgres`:` 5432`

If I execute the command after the container is standing it does not generate an error, my prisma .env:
DATABASE_URL = "postgresql: // postgres: docker @ postgres: 5432 / tecnoanjos-db? Schema = public"

My node service is linked and on the same network:
    links:
        - postgres
    networks:
      - app-network

Details error: Error: P1001: Can't reach database server at `postgres`:`5432`

Please make sure your database server is running at `postgres`:`5432`.
ERROR: Service 'tecno-anjos-backend' failed to build: The command '/bin/sh -c npx prisma migrate save --experimental' returned a non-zero code: 1

**Comment by ryands17** [comment]
Hey @IuryGregoryFaria :wave: 
Could you share your `docker-compose.yml` so that I could look into it?

**Comment by iurygregory-dev** [comment]
version: '3'
services:

  #Nginx Service
  nginx:
    build:
      context: ./nginx/
      dockerfile: Dockerfile
    image: nginx
    container_name: nginx
    restart: always
    tty: true
    ports:
      - "80:80"
      - "443:443"
    volumes:
      - ../.:/var/www/html/backend
      - ./nginx/sites-available/. :/etc/nginx/sites-available/
    networks:
      - app-network

  #Postgres
  postgres:
    image: postgres
    container_name: postgres
    restart: always
    tty: true
    depends_on:
      - nginx
    environment:
      POSTGRES_USER: postgres
      POSTGRES_DB: db
      POSTGRES_PASSWORD: docker
    ports:
      - "5432:5432"
    volumes:
      - dbdata:/var/lib/postgresql/data
    networks:
      - app-network

  #Node Service
  backend:
    build:
      context: ../
      dockerfile: Dockerfile
    image: backend
    container_name: backend
    links:
        - postgres
    restart: always
    tty: true
    environment:
      SERVICE_NAME: backend
    working_dir: /var/www/html/backend
    networks:
      - app-network

#Docker Networks
networks:
  app-network:
    driver: bridge
#Volumes
volumes:
  dbdata:
    driver: local

**Reply by ryands17** [reply]
Are you running `npx prisma migrate save --experimental` outside the Docker container? If so, you would have to use `localhost` instead as `postgres` is only available inside the context of `docker-compose.yml` and not outside.

Also you would have to execute the command after the postgres service has started successfully.

**Comment by iurygregory-dev** [comment, marked as answer]
I'm running it inside the docker file, how can I run it with the docker-compose? since the docker-compose calls the dockerfile that generates the image

**Reply by ryands17** [reply]
If you're running in the Dockerfile, it's possible that Postgres has not started yet, so you would have to add a wait script that waits till postgres has started and then you can run `prisma migrate`.

Something like [this](https://docs.docker.com/compose/startup-order/) that uses a wait command would be helpful.

**Reply by iurygregory-dev** [reply]
thank you very much my friend, I had read about it, but I thought it wouldn't work, it was really worth it, a return hj ate later with this solution!

**Comment by iurygregory-dev** [comment]
hi, I managed to solve the problem using our friend's tip, create command to wait for another service to start on the docker compose, thank you very much

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
